modules/table/src/main/java/org/apache/ignite/internal/table/distributed/LowWatermark.java

```diff
@@ -117,50 +121,47 @@ public LowWatermark(
     /**
      * Starts the watermark manager.
      */
-    public void start() {
-        inBusyLock(busyLock, () -> {
-            vaultManager.get(LOW_WATERMARK_VAULT_KEY)
-                    .thenCompose(vaultEntry -> inBusyLock(busyLock, () -> {
-                        if (vaultEntry == null) {
-                            scheduleUpdateLowWatermarkBusy();
-
-                            return nullCompletedFuture();
-                        }
-
-                        HybridTimestamp lowWatermark = ByteUtils.fromBytes(vaultEntry.value());
-
-                        return txManager.updateLowWatermark(lowWatermark)
-                                .thenApply(unused -> {
-                                    this.lowWatermark.set(lowWatermark);
-
-                                    runGcAndScheduleUpdateLowWatermarkBusy(lowWatermark);
-
-                                    return lowWatermark;
-                                });
-                    }))
-                    .whenComplete((lowWatermark, throwable) -> {
-                        if (throwable != null) {
-                            if (!(throwable instanceof NodeStoppingException)) {
-                                LOG.error("Error getting low watermark", throwable);
+    public CompletableFuture<Void> start() {
```

Comment:
Why asI don't see the need for asynchronous code.

Reply:
But it was here before my changes

Reply:
fixed